src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/decisionmaker/DecisionMakerHandler.java

```diff
@@ -71,6 +82,19 @@ private void handleTimer(Tuple input) {
         service.tick();
     }
 
+    private void handleLifeCycleEvent(Tuple input) {
+        LifecycleEvent event = (LifecycleEvent) input.getValueByField(ZooKeeperSpout.FIELD_ID_LIFECYCLE_EVENT);
+        if (event.getSignal().equals(Signal.SHUTDOWN)) {
+            service.deactivate();
+            emit(STREAM_ZOOKEEPER_ID, input, new Values(event, getCommandContext()));
+        } else if (event.getSignal().equals(Signal.START)) {
+            service.activate();
+            emit(STREAM_ZOOKEEPER_ID, input, new Values(event, getCommandContext()));
+        } else {
+            log.info("Received signal info %s", event.getSignal());
```

Comment:
%s -> {}
I don't know how many times this chunk is copy-pasted, but definitely, there are some more of them.

Reply:
Fixed